### ISSUE TITLE
worpdress-core-installer fails to install with Composer 2.0

### ISSUE THREAD

**Issue by furins** [roots/trellis#1245]
* [X] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [X] This request isn't a duplicate of an existing issue (well, it is. But see below.)
* [X] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description
`worpdress-core-installer` 1.1.0 fails to install with Composer 2.0.
The package revision 1.1.0 is the latest one available on packagist.org; a more recent PR has been already merged into main branch of `roots/worpdress-core-installer` to solve the issue but a new release has not been published yet.
It's a kind of duplicate issue, I know, since the correct solution should take place on the aforementioned package. But this issue is blocking also new provisions of trellis, and can be solved forcing Composer <2.0, in case.

 For more details and reproducibility see: [https://github.com/roots/wordpress-core-installer/issues/4](https://github.com/roots/wordpress-core-installer/issues/4)

**Comment by schuhwerk** [comment]
Hi! I ran into the same issue today provisioning a new server. One problem occured before:

> Package hirak/prestissimo at version @stable has a PHP requirement incompatible with your PHP version, PHP extensions and Composer version

The package is added [here](https://github.com/roots/trellis/blob/master/group_vars/all/main.yml) and requires composer-plugin-api ^1.0.0 ([packagist](https://packagist.org/packages/hirak/prestissimo)).

Adding a version to the `dev.yml`/`server.yml` fixes the issue for me (for now).
```yml
   - { role: composer, tags: [composer], composer_version: '1.*', composer_keep_updated: false }
```

**Comment by furins** [comment]
Hi @schuhwerk, I did exactly the same, it's a safe workaround.
Wordpress-core-installer will be fixed soon, [probably tomorrow](https://github.com/roots/wordpress-core-installer/issues/4). This should allow us to install composer 2.0 and entirely remove the prestissimo requirement (according to what I've read composer 2.0 [already has parallel downloads](https://github.com/hirak/prestissimo/issues/218#issuecomment-610581259)).

**Comment by swalkinshaw** [comment]
1. `roots/wordpress-core-installer` has been updated
2. Bedrock has been updated to use that latest version
3. I'm removing `prestissimo` from Trellis in https://github.com/roots/trellis/pull/1247

**Comment by swalkinshaw** [comment]
For anyone viewing this issue, you have a two main options:

a. update `roots/wordpress-core-installer` to `1.100.0`
b. pin Composer to 1.x as shown [above](https://github.com/roots/trellis/issues/1245#issuecomment-716165875)

And assuming you've updated `roots/wordpress-core-installer`, you should removed prestissimo as well as in #1247

**Comment by dalepgrant** [comment]
Hopefully some useful further info for future readers: 

I just hit this on a brand new server (Ubuntu 20.10), on the very first deploy. The above step (update `roots/wordpress-core-installer` to `1.100.0`) ultimately solved this issue 👌  but I then got hit by an unusual error which took me a while to resolve: 
```
Git repo git@github.com:SmoofCreative/thecambridgecode.com.git cannot be
accessed. Please verify the repository exists and you have SSH forwarding set
up correctly.
```
I thought this was odd, given I'd just managed a deploy. Seems unrelated, but the only thing I'd changed was updating `roots/wordpress-core-installer`. 

Long story short, I had to SSH into the server and delete the `/shared/source` folder to be able to deploy again. I spent a good while checking discourse and confirming SSH forwarding was set up, branch & repo was correct, folder permissions etc. so just leaving this here to save someone else.

**Comment by martinknorr-spacepilots** [comment]
Hey guys I'm feeling super stupid, but how exactly do I update `roots/wordpress-core-installer` to `1.100.0`? @swalkinshaw @dalepgrant

**Comment by josialoos** [comment]
> Hey guys I'm feeling super stupid, but how exactly do I update `roots/wordpress-core-installer` to `1.100.0`? @swalkinshaw @dalepgrant

+1

**Comment by martinknorr-spacepilots** [comment]
Okay, so if I do `vagrant ssh` and go to `cd /srv/www/example.dev/current` I can run `composer update` which did the trick for me

**Comment by TangRufus** [comment]
> how exactly do I update roots/wordpress-core-installer to 1.100.0? 

With composer v2, run `$ composer update -W`
